tensorflow/python/ops/script_ops.py

```diff
@@ -104,8 +104,11 @@ def __call__(self, device, token, args):
     """Passes `args` to `self._func`, which is executed eagerly."""
 
     with context.eager_mode(), backprop.GradientTape() as tape:
+      # Only watch tensors with a floating dtype.
       for tensor in args:
-        tape.watch(tensor)
+        for t in nest.flatten(tensor):
+          if t.dtype.is_floating:
```

Comment:
It looks like we should have also watched complex types here?
(context: https://github.com/tensorflow/tensorflow/issues/32774)

Reply:
This had been fixed in 507325c5b3fa5943485f3f994048fe7683e0f95d